blockly-core/core/ui/fields/field_parameter.js

```diff
@@ -74,10 +74,16 @@ Blockly.FieldParameter.dropdownChange = function(text) {
       Blockly.Variables.renameVariable(oldVar, text, this.sourceBlock_.blockSpace);
     }
   } else if (text === Blockly.Msg.DELETE_PARAMETER) {
-    var result = window.confirm(Blockly.Msg.DELETE_PARAMETER_TITLE.replace('%1', oldVar));
-    if (result) {
-      Blockly.Variables.deleteVariable(oldVar, this.sourceBlock_.blockSpace);
-    }
+    Blockly.showSimpleDialog({
+      bodyText: Blockly.Msg.DELETE_PARAMETER_TITLE.replace('%1', oldVar),
+      cancelText: Blockly.Msg.DELETE,
+      confirmText: Blockly.Msg.KEEP,
+      onConfirm: null,
+      onCancel: function() {
```

Comment:
oh, great!

I had not noticed that, but I believe it; our usage of dialogs is pretty messy, particularly given the mix of templating strategies they're using. If you have time this sprint, I'd love to see a renaming pass!